src/Button.js

```diff
@@ -2,38 +2,53 @@ import PropTypes from 'prop-types'
 import styled from 'styled-components'
 import {COMMON, get} from './constants'
 import theme from './theme'
-import getButtonStyles from './ButtonStyles'
-import {layout} from 'styled-system'
+import buttonStyles from './ButtonStyles'
+import {compose, system, layout} from 'styled-system'
 import systemPropTypes from '@styled-system/prop-types'
 
-function fontSize({size = '14px', ...props}) {
-  return {
-    fontSize:
-      size === 'sm' ? `${get('fontSizes.0')(props)}px` : size === 'large' ? `${get('fontSizes.2')(props)}px` : size
-  }
+const fontSizeVariants = {
+  small: get('fontSizes.0'),
+  medium: get('fontSizes.1'),
+  large: get('fontSizes.2')
 }
 
-const Button = styled.button.attrs(props => ({
-  onClick: props.disabled ? undefined : props.onClick,
-  className: props.disabled ? 'disabled' : ''
+const variants = ({variant, theme}) => {
+  const getter = fontSizeVariants[variant] || fontSizeVariants['medium']
+  return {fontSize: getter(theme)}
+}
+
+const Button = styled.button.attrs(({disabled, onClick}) => ({
+  onClick: disabled ? undefined : onClick
 }))`
-  ${props => (props.theme ? getButtonStyles(props.theme) : '')};
-  ${fontSize};
-  ${COMMON};
-  ${layout};
+  ${buttonStyles}
+  ${variants}
+  ${compose(
+    system({
+      fontSize: {
+        property: 'fontSize',
+        cssProperty: 'fontSize',
+        scale: 'fontSizes'
+      }
+    }),
+    COMMON,
+    layout
+  )}
 `
 
 Button.defaultProps = {
-  theme
+  fontSize: undefined,
```

Comment:
for cases where we are setting the default to undefined I think that we can just omit them from `defaultProps` alltogether